src/types.c

```diff
@@ -133,6 +136,11 @@ bool is_number_type(const Type *t)
 {
     return is_integer_type(t) || t->kind == TYPE_DOUBLE;
```

Comment:
Floats are numbers too:
```suggestion
    return is_integer_type(t) || t->kind == TYPE_DOUBLE || t->kind == TYPE_FLOAT;
```

Reply:
fixed